code/game/turfs/open.dm

```diff
@@ -257,3 +257,122 @@
 
 /turf/open/shuttle/elevator/grating
 	icon_state = "floor_grating"
+
+// LAVA
+
+/turf/open/lavaland
+	icon = 'icons/turf/lava.dmi'
+	plane = FLOOR_PLANE
+	baseturfs = /turf/open/lavaland/lava
+
+/turf/open/lavaland/lava
+	name = "lava"
+	icon_state = "full"
+
+	light_range = 4
+	light_power = 0.75
+	light_color = LIGHT_COLOR_LAVA
+
+/turf/open/lavaland/lava/is_weedable()
+	return FALSE
+
+/turf/open/lavaland/lava/corner
+	icon_state = "corner"
+
+/turf/open/lavaland/lava/side
+	icon_state = "side"
+
+/turf/open/lavaland/lava/lpiece
+	icon_state = "lpiece"
+
+/turf/open/lavaland/lava/New()
+	..()
+
+/turf/open/lavaland/lava/Entered(atom/movable/AM)
+	if(burn_stuff(AM))
+		START_PROCESSING(SSobj, src)
+
+/turf/open/lavaland/lava/Exited(atom/movable/Obj, atom/newloc)
+	. = ..()
+	if(isliving(Obj))
+		var/mob/living/L = Obj
+		if(!islava(newloc) && !L.on_fire)
+			L.update_fire()
+
+/turf/open/lavaland/lava/process()
+	if(!burn_stuff())
+		STOP_PROCESSING(SSobj, src)
+
+/turf/open/lavaland/lava/proc/burn_stuff(AM)
+	. = 0
+
+	var/thing_to_check = src
+	if (AM)
+		thing_to_check = list(AM)
+	for(var/thing in thing_to_check)
+		if(isobj(thing))
+			var/obj/O = thing
+			O.fire_act(10000, 1000)
+        
+		else if (isliving(thing))
+			var/mob/living/L = thing
+			if(L.stat == DEAD)
+				continue
+	
+			if(!L.on_fire || L.getFireLoss() <= 200)
+				L.adjustFireLoss(20)
+				L.adjust_fire_stacks(20)
+				L.IgniteMob()
```

Comment:
This section of the code which does the damage to mobs doesn't respect the mobs armor.

My recommendation is to instead use the proc `take_overall_damage`

`take_overall_damage(brute, burn, blocked = 0, sharp = FALSE, edge = FALSE, updating_health = FALSE)`

On humans this affects limbs at random.

So you would set the arguments to `(0, 20, L.getSoftArmor(fire))` however I'm not certain on precisely how to get the targets armor value at this point in time.

Reply:
I think my only issue with this is, is that I dont want marines with high fire protection to have the ability to walk on the lava. That would make it extremely strong.

Reply:
Could make it half of the soft armor value then.

Reply:
hmm, true.

Reply:
you could use `clamp` here

Reply:
I believe I have done this.